src/line_cursor.ts

```diff
@@ -475,6 +475,43 @@ export class LineCursor extends Marker {
     throw new Error('no valid nodes in this.potentialNodes');
   }
 
+  /**
+   * Get the current location of the cursor.
+   *
+   * Overrides normal Marker getCurNode to update the current node from the selected
+   * block. We typically update the node from the selection via a listener but
+   * that is not called immediately when `Gesture` calls `Blockly.common.setSelected`.
+   * In particular the listener runs after showing the context menu.
+   *
+   * @returns The current field, connection, or block the cursor is on.
+   */
+  override getCurNode(): ASTNode | null {
+    const curNode = super.getCurNode();
+    let selected = Blockly.common.getSelected();
+    if (
+      selected === null &&
+      curNode?.getType() === Blockly.ASTNode.types.BLOCK
+    ) {
+      // Selection says our curNode is not selected anymore.
+      // this.setCurNode(null as never, true);
```

Comment:
Oops, this is definitely a problem. Re-enabling it threw up a shadow block issue addressed in 1233f578fd94a0941a8e028a0417d7e83a3eda3f

Reply:
I actually missed this commented out bit the first time around--great catch and thanks for fixing it in the follow-up!